netstandard/tools/HandlePackageFileConflicts.cs

```diff
@@ -0,0 +1,323 @@
+            var conflicts = new HashSet<ITaskItem>();
+
+            foreach (var item in items)
+            {
+                var targetPath = GetTargetPath(item);
+
+                ITaskItem existingItem;
+
+                if (targetPathToItem.TryGetValue(targetPath, out existingItem))
+                {
+                    // a conflict was found, determine the winner.
+                    var winner = HandleConflict(existingItem, item);
+
+                    if (winner == null)
+                    {
+                        // no winner, skip it.
+                        // don't add to conflict list and just use the existing item for future conflicts.
+                        continue;
+                    }
+
+                    if (winner != existingItem)
+                    {
+                        // replace existing item and add it to conflict list.
+                        targetPathToItem[targetPath] = item;
+                        conflicts.Add(existingItem);
+                    }
+                    else
+                    {
+                        // no need to replace item, just add new item to conflict list.
+                        conflicts.Add(item);
+                    }
+                }
+                else
+                {
+                    targetPathToItem[targetPath] = item;
+                }
+            }
+
+            return RemoveConflicts(items, conflicts);
+        }
+
+        private Version GetFileVersion(string sourcePath)
+        {
+            var fvi = FileVersionInfo.GetVersionInfo(sourcePath);
+
+            if (fvi != null)
+            {
+                return new Version(fvi.FileMajorPart, fvi.FileMinorPart, fvi.FileBuildPart, fvi.FilePrivatePart);
+            }
+
+            return null;
+        }
+
+        private int GetPackageRank(ITaskItem item)
+        {
+            int rank = int.MaxValue;
+
+            // NuGet 3
+            var packageId = item.GetMetadata("NuGetPackageId");
+
+            if (String.IsNullOrWhiteSpace(packageId))
+            {
+                // NuGet 4
+                packageId = item.GetMetadata("ParentPackage");
+
+                var versionSeperatorIndex = packageId.IndexOf('/');
+
+                if (versionSeperatorIndex != -1)
+                {
+                    packageId = packageId.Substring(0, versionSeperatorIndex);
+                }
+            }
+
+            if (!String.IsNullOrWhiteSpace(packageId) && PreferredPackages != null)
+            {
+                EnsurePackageRanks();
+
+                packageRanks.TryGetValue(packageId, out rank);
+            }
+
+            return rank;
+        }
+
+        private static string GetSourcePath(ITaskItem item)
+        {
+            var sourcePath = item.GetMetadata("HintPath");
+
+            if (String.IsNullOrWhiteSpace(sourcePath))
+            {
+                sourcePath = item.ItemSpec;
+            }
+
+            return sourcePath;
```

Comment:
We don't need to normalize source, only target path.  We do normalize in GetTargetPath